extensions/lifecycle/tasks/LifecycleUpdateExpirationTask.js

```diff
@@ -0,0 +1,169 @@
+            key,
+            version,
+        } = entry.getAttribute('target');
+
+        const backbeatClient = this.getBackbeatMetadataProxy(canonicalId, accountId);
+        if (!backbeatClient) {
+            log.error('failed to get backbeat client', {
+                canonicalId,
+                accountId,
+            });
+            done(errors.InternalError.customizeDescription(
+                'Unable to obtain client',
+            ));
+            return;
+        }
+
+        backbeatClient.putMetadata({
+            bucket,
+            objectKey: key,
+            versionId: version,
+            mdBlob: objMD.getSerialized(),
+        }, log, err => {
+            if (err) {
+                log.error(
+                    'an error occurred when updating metadata for transition',
+                    Object.assign(
+                        {
+                            method: 'LifecycleUpdateExpirationTask._putMetadata',
+                            error: err.message,
+                        },
+                        entry.getLogInfo(),
+                    )
+                );
+                done(err);
+                return;
+            } else {
+                log.end().info('metadata updated for transition', entry.getLogInfo());
+                done();
+                return;
+            }
+        });
+    }
+
+    _garbageCollectLocation(entry, locations, log, done) {
+        const { bucket, key, version, eTag } = entry.getAttribute('target');
+        const gcEntry = ActionQueueEntry.create('deleteData')
+              .addContext({
+                  origin: 'lifecycle',
+                  ruleType: 'restore',
+                  reqId: log.getSerializedUids(),
+                  bucketName: bucket,
+                  objectKey: key,
+                  versionId: version,
+                  eTag,
+              })
+              .setAttribute('source', entry.getAttribute('source'))
+              .setAttribute('serviceName', 'lifecycle-transition')
+              .setAttribute('target.locations', locations);
+        this.gcProducer.publishActionEntry(gcEntry);
+        return process.nextTick(done);
+    }
+
+    /**
+     * Execute the action specified in action entry to update expirations on an object
+     *
+     * @param {ActionQueueEntry} entry - action entry to execute
+     * @param {Function} done - callback funtion
+     * @return {undefined}
+     */
+    processActionEntry(entry, done) {
+        const log = this.logger.newRequestLogger();
+        entry.addLoggedAttributes({
+            bucketName: 'target.bucket',
+            objectKey: 'target.key',
+            versionId: 'target.version',
+        });
+
+        async.waterfall([
+            next => this._getMetadata(entry, log, next),
+            (objMD, next) => {
+                const archive = objMD.getArchive();
+                objMD.setArchive({
+                    archiveInfo: archive.archiveInfo,
+                });
```

Comment:
this code looks/reads like a NOOP, while it does the reset all the "time" fields : which I imagine is what is actually useful here....

so i thing this code should be made more explicit about what it does : at minimum, adding a callback expliciting that we are resetting these fields ; or better yet introduce a function whose goal is exactly this...

For exemple, something like this (where `resetRestoration()` may actually have the same implementation you put here) would make it much easier to read IMHO:

```suggestion
                const archive = objMD.getArchive();
                archive.resetRestoration();
```

Reply:
I could do that in a second pass as the object MD is part of Arsenal. 
It could be possible to do `objMD.resetRestoration();` directly instead of getting the archive first.